### PR TITLE
Map voting system

### PR DESCRIPTION
**Pull request by tarnas14**
- added handler for listing maps
- added ability to vote up/down a map for a specific keyword
- added a parameter for newladder handler to affect map selection
- grouped map-related functionality inside a separate folder (voting handlers, listing handler and map selection strategy)
- newladder now favours maps with higher vote count for specified keyword (optional)
- udpated readme with descriptions of all handlers (new map handlers as well as old handlers)

### CODE REVIEW

**Comment by tarnas14** [comment]
I was thinking about restricting voting to one vote per player, but this would limit the 'change of heart' factor - if someone grows to like a map after they'd downvoted it they can now just upvote it twice to make a difference (which would not be possible if we restricted voting to 1/player)
the drawback is that now you can spam voting system like an idiot, but I think we can assume people don't want to break the system they are using, right?

File: README.md [hunk 1]
```diff
@@ -4,22 +4,56 @@ Klesk is a Slack service to store and return ladder data for Quake Live or whate
 [![Build Status](https://travis-ci.org/TheOneYouDontLike/Klesk.svg?branch=master)](https://travis-ci.org/TheOneYouDontLike/Klesk)
 
 ## Features
-- Creating new ladder
-- Joining existing ladder
-- Adding match result
-- Showing user statistics for ladder
-- Showing ranking
+- `newladder <laddername> [<keyword>]`<br/>
```
Comment (Wojtasik) [inline README.md:7]: Do you need these `<br />`'s?

File: app/maps/mapSelection.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            biasedMapList.push(map);
+        }
+    });
+
+    biasedMapList = _.shuffle(biasedMapList);
+
+    return biasedMapList;
+}
+
+function _initialiseVotesForMapsWithoutKeywordVote(maps, keyword) {
```
Comment (Wojtasik) [inline app/maps/mapSelection.js:21]: Are we using british or american spelling? It's about "initialise".

File: app/maps/mapSelection.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+import _ from 'lodash';
+
+function _getMapListFavouringUpvotedMaps(maps, keyword) {
+    let biasedMapList = [];
```
Comment (Wojtasik) [inline app/maps/mapSelection.js]: Are we using british or american spelling? It's about biased.

File: app/maps/mapVoteHandlers.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+function _getMapVoteHandler(mapPersistence, voteStrategy) {
+    return {
+        makeItSo(parsedCommand, callback) {
+            if (parsedCommand.arguments.length < 2) {
+                callback(null, 'Map name and keyword is required for voting.');
+                return;
+            }
+
+            if (parsedCommand.arguments.length < 3) {
+                callback(null, 'Keyword is required for voting.');
```
Comment (Wojtasik) [inline app/maps/mapVoteHandlers.js]: What do you think about adding correct command as example to the message (and maybe other)? User can add keyword without map name and this message will be confusing.

**Comment by Wojtasik** [comment]
I'm not sure about people behaviour, be leave it as it is. We'll see what comes out. ;)
Anyway, it looks good. Let's wait for Prezes comments.

File: README.md (hunk 1 above)
Comment (tarnas14) [inline README.md:7]: it looks better - without them markdown doesn't give you line break, but if you put two linebreaks then there's an empty line between lines of text :d

File: app/maps/mapSelection.js (hunk 2 above)
Comment (tarnas14) [inline app/maps/mapSelection.js:21]: err I don't know, might change it if you don't like this (PROPER) spelling

File: app/maps/mapSelection.js (hunk 3 above)
Comment (tarnas14) [inline app/maps/mapSelection.js]: I've looked it up, apparently you can spell it 'biassed' (double s), but I don't think people under the age of 120 use that spelling anymore ;D

File: app/maps/mapVoteHandlers.js (hunk 4 above)
Comment (tarnas14) [inline app/maps/mapVoteHandlers.js]: you're right, we could add suggestion with the full syntax in the message, like `upvotemap <mapname> <keyword>`

**Comment by tarnas14** [comment]
I had an idea about maybe adding 'disabled commands' config or something - you could then disable the voting commands and have the administrator tweak the map keyword numbers :)

**Pull request merged**